src/script/media/MediaDevicesHandler.ts

```diff
@@ -265,31 +270,32 @@ export class MediaDevicesHandler {
       }
     };
 
-    return navigator.mediaDevices
-      .enumerateDevices()
-      .catch(error => {
-        this.logger.error(`Failed to update MediaDevice list: ${error.message}`, error);
-        throw error;
-      })
-      .then(mediaDevices => {
-        this.removeAllDevices();
-
-        if (mediaDevices) {
-          const filteredDevices = this.filterMediaDevices(mediaDevices);
-
-          setDevices(MediaDeviceType.AUDIO_INPUT, filteredDevices.microphones);
-          setDevices(MediaDeviceType.AUDIO_OUTPUT, filteredDevices.speakers);
-          setDevices(MediaDeviceType.VIDEO_INPUT, filteredDevices.cameras);
-          this.previousDeviceSupport = {
-            audioinput: filteredDevices.microphones.length,
-            audiooutput: filteredDevices.speakers.length,
-            videoinput: filteredDevices.cameras.length,
-          };
-          return mediaDevices;
-        }
+    try {
+      this.removeAllDevices();
+      const mediaDevices = await navigator.mediaDevices.enumerateDevices();
 
+      if (!mediaDevices) {
         throw new Error('No media devices found');
-      });
+      }
+
+      const filteredDevices = this.filterMediaDevices(mediaDevices);
+
+      setDevices(MediaDeviceType.AUDIO_INPUT, filteredDevices.microphones);
+      setDevices(MediaDeviceType.AUDIO_OUTPUT, filteredDevices.speakers);
+      setDevices(MediaDeviceType.VIDEO_INPUT, filteredDevices.cameras);
+      this.previousDeviceSupport = {
+        audioinput: filteredDevices.microphones.length,
+        audiooutput: filteredDevices.speakers.length,
+        videoinput: filteredDevices.cameras.length,
+      };
+
+      this.onMediaDevicesRefresh?.();
```

Comment:
This is the actual fixin this method, the rest is just refactor from then().catch() to try/catch block